Tools/build/check_warnings.py

```diff
@@ -43,35 +82,25 @@ def extract_warnings_from_compiler_output(compiler_output: str) -> list[dict]:
 
 def get_warnings_by_file(warnings: list[dict]) -> dict[str, list[dict]]:
     """
-    Returns a dictionary where the key is the file and the data is the warnings
-    in that file
+    Returns a dictionary where the key is the file and the data is the warnings in that file
     """
     warnings_by_file = {}
```

Comment:
Perhaps replace this with a defaultdict?

Up top:

```python
from collections import defaultdict
```

```suggestion
    warnings_by_file = defaultdict(list)
```

Reply:
Good idea. Switched to defaultdict to avoid checking if keys exist